hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/EthereumTransactionHandler.java

```diff
@@ -0,0 +1,91 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.contract.impl.handlers;
+
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.hapi.node.base.HederaFunctionality;
+import com.hedera.node.app.service.contract.impl.exec.TransactionComponent;
+import com.hedera.node.app.service.contract.impl.infra.EthereumCallDataHydration;
+import com.hedera.node.app.service.contract.impl.infra.EthereumSignatures;
+import com.hedera.node.app.service.contract.impl.records.EthereumTransactionRecordBuilder;
+import com.hedera.node.app.service.file.ReadableFileStore;
+import com.hedera.node.app.spi.workflows.HandleContext;
+import com.hedera.node.app.spi.workflows.HandleException;
+import com.hedera.node.app.spi.workflows.PreHandleContext;
+import com.hedera.node.app.spi.workflows.TransactionHandler;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+/**
+ * This class contains all workflow-related functionality regarding {@link
+ * HederaFunctionality#ETHEREUM_TRANSACTION}.
+ */
+@Singleton
+public class EthereumTransactionHandler implements TransactionHandler {
+    private final EthereumSignatures ethereumSignatures;
+    private final EthereumCallDataHydration callDataHydration;
+    private final Provider<TransactionComponent.Factory> provider;
+
+    @Inject
+    public EthereumTransactionHandler(
+            @NonNull final EthereumSignatures ethereumSignatures,
+            @NonNull final EthereumCallDataHydration callDataHydration,
+            @NonNull final Provider<TransactionComponent.Factory> provider) {
+        this.ethereumSignatures = requireNonNull(ethereumSignatures);
+        this.callDataHydration = requireNonNull(callDataHydration);
+        this.provider = requireNonNull(provider);
+    }
+
+    @Override
+    public void preHandle(@NonNull final PreHandleContext context) {
+        requireNonNull(context);
+        final var body = context.body().ethereumTransactionOrThrow();
+        final var fileStore = context.createStore(ReadableFileStore.class);
+        final var ethTxData = callDataHydration.tryToHydrate(body, fileStore).ethTxData();
+        if (ethTxData != null) {
+            // Ignore the return value; we just want to cache the signature for use in handle()
+            ethereumSignatures.impliedBy(ethTxData);
```

Comment:
Is ethereumSignatures the best name for this cache?  I think `signatures` threw me off.